plugins/ssh-plugin/src/ssh-plugin-backend.ts

```diff
@@ -307,6 +323,61 @@ const deleteKeyPair = async (sshkeyManager: SshKeyManager) => {
     }
 };
 
+async function registerKey(keyPath: string, passphrase: string): Promise<void> {
+    try {
+        await sshAdd(keyPath, passphrase);
+    } catch (e) {
+        if (e.includes('Could not open a connection to your authentication agent')) {
+            await startSshAgent();
+            await sshAdd(keyPath, passphrase);
+        }
+    }
+}
+
+async function isEncrypted(keyPath: string): Promise<boolean> {
+    return new Promise<boolean>(resolvePromise => {
+        const command = spawn('sshpass', ['-p', '', '-P', 'assphrase', 'ssh-keygen', '-y', '-f', keyPath]);
+        command.stdout.on('data', data => {
+            resolvePromise(false);
+        });
+        command.stderr.on('data', data => {
+            if (data.includes('incorrect passphrase supplied to decrypt private key')) {
+                resolvePromise(true);
+            }
+        });
+    });
+}
+
+function sshAdd(keyPath: string, passphrase: string): Promise<void> {
+    return new Promise<void>((resolvePromise, reject) => {
+        const command = spawn('sshpass', ['-p', passphrase, '-P', 'assphrase', 'ssh-add', keyPath]);
```

Comment:
is the second parameter really required?
`-P assphrase` - looks strange )
typo?

Reply:
It's not a typo. It is needed to change the prompt type https://stackoverflow.com/a/57290159

Reply:
Otherwise the command will freeze :)